src/daft-catalog/src/session.rs

```diff
@@ -0,0 +1,51 @@
+use daft_logical_plan::LogicalPlanBuilder;
+
+use crate::{errors::Result, identifier::Identifier, DaftCatalog};
+
+/// Temporary rename while refactoring.
+type Metastore = DaftCatalog;
+
+/// Session holds all state for query planning and execution (e.g. connection).
+pub struct Session {
+    /// Session identifier
+    pub id: String,
+    // TODO remove DaftCatalog after all APIs are migrated.
+    pub metastore: DaftCatalog,
+    // TODO execution context
+    // TODO session options
+}
+
+impl Session {
+    /// Creates a new empty session
+    pub fn new(id: &str, metastore: Metastore) -> Self {
+        Self {
+            id: id.to_string(),
+            metastore,
+        }
+    }
+
+    /// Creates a table backed by the view
+    /// TODO support names/namespaces and other table sources.
+    pub fn create_table(&mut self, name: &str, view: impl Into<LogicalPlanBuilder>) -> Result<()> {
```

Comment:
for ergonomics, It'd be nice to take anything that impl's `AsRef<str>` for the name. 

```rs
fn create_table<S: AsRef<str>>(&self, name: S, ...)
```

Reply:
I wont update this only because this will become `create_table(&self, name: &Name, ..)` soon since names have different semantics and we need to handle qualifiers.